### PR TITLE
[directxmesh, directxtex, directxtk(12), uvatlas] April 2023 updates

### PR DESCRIPTION
**Pull request by walbourn**
Update for minor servicing releases on GitHub for **directxtex**, **directxmesh**, **directxtk**, **directxtk12**. and **uvatlas**.

https://github.com/microsoft/DirectXMesh/releases/tag/apr2023
https://github.com/microsoft/DirectXTex/releases/tag/apr2023
https://github.com/microsoft/DirectXTK/releases/tag/apr2023
https://github.com/microsoft/DirectXTK12/releases/tag/apr2023
https://github.com/microsoft/UVAtlas/releases/tag/apr2023

> Removed the 'no-omp' feature from DirectXTex introduced in PR 30685 as the upstream CMake has been updated to use a better OpenMP support detection solution. Both DirectXTex and UVAtlas can now build with OpenMP with MVSC, clang, and MinGW/GCC.

### CODE REVIEW

File: ports/uvatlas/0001-Update-CMake-to-use-OpenMP-via-built-in-support.patch [hunk 1]
```diff
@@ -0,0 +1,44 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (walbourn) [inline ports/uvatlas/0001-Update-CMake-to-use-OpenMP-via-built-in-support.patch:1]: These patches are for fixes applied upstream in [this PR](https://github.com/microsoft/UVAtlas/pull/112) to apply the updated OpenMP logic to **uvatlas** as well.

**Comment by walbourn** [comment]
Validated all supported triplets and features:

```
vcpkg install directxmesh:x86-windows
vcpkg install directxmesh[tools]:x86-windows --recurse
vcpkg install directxmesh[dx12]:x86-windows --recurse
vcpkg install directxmesh[spectre]:x86-windows --recurse
vcpkg install directxmesh:x86-windows-static
vcpkg install directxmesh:x86-windows-static-md
vcpkg install directxmesh:x64-windows
vcpkg install directxmesh[tools]:x64-windows --recurse
vcpkg install directxmesh[dx12]:x64-windows --recurse
vcpkg install directxmesh[spectre]:x64-windows --recurse
vcpkg install directxmesh:x64-windows-static
vcpkg install directxmesh:x64-windows-static-md
vcpkg install directxmesh:arm-windows
vcpkg install directxmesh[tools]:arm-windows --recurse
vcpkg install directxmesh[spectre]:arm-windows --recurse
vcpkg install directxmesh:arm64-windows
vcpkg install directxmesh[tools]:arm64-windows --recurse
vcpkg install directxmesh[spectre]:arm64-windows --recurse
vcpkg install directxmesh:arm64-windows-static
vcpkg install directxmesh:arm64-windows-static-md

vcpkg install directxmesh:x86-uwp
vcpkg install directxmesh:x64-uwp
vcpkg install directxmesh:arm-uwp
vcpkg install directxmesh:arm64-uwp

vcpkg install directxmesh:x64-xbox-scarlett
vcpkg install directxmesh:x64-xbox-scarlett-static
vcpkg install directxmesh:x64-xbox-xboxone
vcpkg install directxmesh:x64-xbox-xboxone-static

./vcpkg install directxmesh:x64-linux
./vcpkg install directxmesh[dx12]:x64-linux --recurse

./vcpkg install directxmesh:arm64-linux
./vcpkg install directxmesh[dx12]:arm64-linux --recurse

vcpkg install directxmesh:x64-mingw-static
vcpkg install directxmesh[dx12]:x64-mingw-static --recurse
vcpkg install directxmesh[tools]:x64-mingw-static --recurse

vcpkg install directxmesh:x86-mingw-static
vcpkg install directxmesh[dx12]:x86-mingw-static --recurse
vcpkg install directxmesh[tools]:x86-mingw-static --recurse
```

```
vcpkg install directxtex[core]:x86-windows
vcpkg install directxtex:x86-windows --recurse
vcpkg install directxtex[tools]:x86-windows --recurse
vcpkg install directxtex[dx12]:x86-windows --recurse
vcpkg install directxtex[openexr,tools]:x86-windows --recurse
vcpkg install directxtex[spectre]:x86-windows --recurse
vcpkg install directxtex:x86-windows-static
vcpkg install directxtex:x86-windows-static-md
vcpkg install directxtex[core]:x64-windows
vcpkg install directxtex:x64-windows --recurse
vcpkg install directxtex[tools]:x64-windows --recurse
vcpkg install directxtex[dx12]:x64-windows --recurse
vcpkg install directxtex[openexr,tools]:x64-windows --recurse
vcpkg install directxtex[spectre]:x64-windows --recurse
vcpkg install directxtex:x64-windows-static
vcpkg install directxtex:x64-windows-static-md
vcpkg install directxtex:arm-windows
vcpkg install directxtex[tools]:arm-windows --recurse
vcpkg install directxtex[spectre]:arm-windows --recurse
vcpkg install directxtex:arm64-windows
vcpkg install directxtex[tools]:arm64-windows --recurse 
vcpkg install directxtex[spectre]:arm64-windows --recurse
vcpkg install directxtex:arm64-windows-static
vcpkg install directxtex:arm64-windows-static-md

vcpkg install directxtex:x86-uwp
vcpkg install directxtex:x64-uwp
vcpkg install directxtex:arm-uwp
vcpkg install directxtex:arm64-uwp

vcpkg install directxtex:x64-xbox-scarlett
vcpkg install directxtex:x64-xbox-scarlett-static
vcpkg install directxtex:x64-xbox-xboxone
vcpkg install directxtex:x64-xbox-xboxone-static

./vcpkg install directxtex:x64-linux
./vcpkg install directxtex[dx12]:x64-linux --recurse
./vcpkg install directxtex[openexr]:x64-linux --recurse

./vcpkg install directxtex:arm64-linux
./vcpkg install directxtex[dx12]:arm64-linux --recurse
./vcpkg install directxtex[openexr]:arm64-linux --recurse

vcpkg install directxtex:x64-mingw-static
vcpkg install directxtex[dx12]:x64-mingw-static --recurse
vcpkg install directxtex[tools]:x64-mingw-static --recurse

vcpkg install directxtex:x86-mingw-static
vcpkg install directxtex[dx12]:x86-mingw-static --recurse
vcpkg install directxtex[tools]:x86-mingw-static --recurse
```

```
vcpkg install directxtk:x86-windows
vcpkg install directxtk[tools]:x86-windows --recurse
vcpkg install directxtk[xaudio2-8]:x86-windows --recurse
vcpkg install directxtk[xaudio2-9]:x86-windows --recurse
vcpkg install directxtk[xaudio2redist]:x86-windows --recurse
vcpkg install directxtk[xaudio2redist]:x86-windows-static-md --recurse
vcpkg install directxtk[spectre]:x86-windows --recurse
vcpkg install directxtk:x86-windows-static
vcpkg install directxtk:x86-windows-static-md
vcpkg install directxtk:x64-windows
vcpkg install directxtk[tools]:x64-windows --recurse
vcpkg install directxtk[xaudio2-8]:x64-windows --recurse
vcpkg install directxtk[xaudio2-9]:x64-windows --recurse
vcpkg install directxtk[xaudio2redist]:x64-windows --recurse
vcpkg install directxtk[xaudio2redist]:x64-windows-static-md --recurse
vcpkg install directxtk[spectre]:x64-windows --recurse
vcpkg install directxtk:x64-windows-static
vcpkg install directxtk:x64-windows-static-md
vcpkg install directxtk:arm-windows
vcpkg install directxtk[tools]:arm-windows --recurse
vcpkg install directxtk[spectre]:arm-windows --recurse
vcpkg install directxtk:arm64-windows
vcpkg install directxtk[tools]:arm64-windows --recurse
vcpkg install directxtk[spectre]:arm64-windows --recurse
vcpkg install directxtk:arm64-windows-static
vcpkg install directxtk:arm64-windows-static-md

vcpkg install directxtk:x86-uwp
vcpkg install directxtk:x64-uwp
vcpkg install directxtk:arm-uwp
vcpkg install directxtk:arm64-uwp

vcpkg install directxtk:x64-mingw-static
vcpkg install directxtk[xaudio2redist]:x64-mingw-static --recurse
vcpkg install directxtk[tools]:x64-mingw-static --recurse

vcpkg install directxtk:x86-mingw-static
vcpkg install directxtk[xaudio2redist]:x86-mingw-static --recurse
vcpkg install directxtk[tools]:x86-mingw-static --recurse
```

```
vcpkg install directxtk12[core]:x86-windows
vcpkg install directxtk12[xaudio2-9]:x86-windows --recurse
vcpkg install directxtk12[xaudio2redist]:x86-windows --recurse
vcpkg install directxtk12[xaudio2redist]:x86-windows-static-md --recurse
vcpkg install directxtk12[spectre]:x86-windows-static-md --recurse
vcpkg install directxtk12:x86-windows-static
vcpkg install directxtk12:x86-windows-static-md
vcpkg install directxtk12[core]:x64-windows
vcpkg install directxtk12[tools]:x64-windows --recurse
vcpkg install directxtk12[xaudio2-9]:x64-windows --recurse
vcpkg install directxtk12[xaudio2redist]:x64-windows --recurse
vcpkg install directxtk12[xaudio2redist]:x64-windows-static-md --recurse
vcpkg install directxtk12[spectre]:x64-windows-static-md --recurse
vcpkg install directxtk12:x64-windows-static
vcpkg install directxtk12:x64-windows-static-md
vcpkg install directxtk12:arm-windows
vcpkg install directxtk12[spectre]:arm-windows --recurse
vcpkg install directxtk12[core]:arm64-windows
vcpkg install directxtk12[tools]:arm64-windows --recurse
vcpkg install directxtk12[spectre]:arm64-windows --recurse
vcpkg install directxtk12:arm64-windows-static
vcpkg install directxtk12:arm64-windows-static-md

vcpkg install directxtk12:x86-uwp
vcpkg install directxtk12:x64-uwp
vcpkg install directxtk12:arm-uwp
vcpkg install directxtk12:arm64-uwp

vcpkg install directxtk12:x64-xbox-scarlett
vcpkg install directxtk12:x64-xbox-scarlett-static
vcpkg install directxtk12:x64-xbox-xboxone
vcpkg install directxtk12:x64-xbox-xboxone-static

vcpkg install directxtk12:x64-mingw-static
vcpkg install directxtk12[xaudio2redist]:x64-mingw-static --recurse
vcpkg install directxtk12[tools]:x64-mingw-static --recurse

vcpkg install directxtk12:x86-mingw-static
vcpkg install directxtk12[xaudio2redist]:x86-mingw-static --recurse
```

```
vcpkg install uvatlas:x86-windows
vcpkg install uvatlas[tools]:x86-windows --recurse
vcpkg install uvatlas[eigen]:x86-windows --recurse
vcpkg install uvatlas[spectre]:x86-windows --recurse
vcpkg install uvatlas:x86-windows-static
vcpkg install uvatlas:x86-windows-static-md
vcpkg install uvatlas:x64-windows
vcpkg install uvatlas[tools]:x64-windows --recurse
vcpkg install uvatlas[eigen]:x64-windows --recurse
vcpkg install uvatlas[spectre]:x64-windows --recurse
vcpkg install uvatlas:x64-windows-static
vcpkg install uvatlas:x64-windows-static-md
vcpkg install uvatlas:arm-windows
vcpkg install uvatlas[tools]:arm-windows --recurse
vcpkg install uvatlas[spectre]:arm-windows --recurse
vcpkg install uvatlas:arm64-windows
vcpkg install uvatlas[tools]:arm64-windows --recurse
vcpkg install uvatlas[spectre]:arm64-windows --recurse
vcpkg install uvatlas:arm64-windows-static
vcpkg install uvatlas:arm64-windows-static-md

vcpkg install uvatlas:x86-uwp
vcpkg install uvatlas:x64-uwp
vcpkg install uvatlas:arm-uwp
vcpkg install uvatlas:arm64-uwp

vcpkg install uvatlas:x64-xbox-scarlett
vcpkg install uvatlas:x64-xbox-scarlett-static
vcpkg install uvatlas:x64-xbox-xboxone
vcpkg install uvatlas:x64-xbox-xboxone-static

./vcpkg install uvatlas:x64-linux 
./vcpkg install uvatlas[eigen]:x64-linux  --recurse

./vcpkg install uvatlas:arm64-linux
./vcpkg install uvatlas[eigen]:arm64-linux --recurse

vcpkg install uvatlas:x64-mingw-static
vcpkg install uvatlas[tools]:x64-mingw-static --recurse

vcpkg install uvatlas:x86-mingw-static
vcpkg install uvatlas[tools]:x86-mingw-static --recurse
```

**Comment by walbourn** [comment]
Also validated against the proposed [clangcl triplets](https://github.com/microsoft/vcpkg/pull/31028).

```
vcpkg install directxmesh:arm64-clangcl-static
vcpkg install directxtex:arm64-clangcl-static
vcpkg install uvatlas:arm64-clangcl-static
vcpkg install directxtk:arm64-clangcl-static
vcpkg install directxtk12:arm64-clangcl-static

vcpkg install directxmesh:arm64-clangcl-dynamic
vcpkg install directxtex:arm64-clangcl-dynamic
vcpkg install uvatlas:arm64-clangcl-dynamic
vcpkg install directxtk:arm64-clangcl-dynamic
vcpkg install directxtk12:arm64-clangcl-dynamic

vcpkg install directxmesh:x64-clangcl-static
vcpkg install directxtex:x64-clangcl-static
vcpkg install uvatlas:x64-clangcl-static
vcpkg install directxtk:x64-clangcl-static
vcpkg install directxtk12:x64-clangcl-static

vcpkg install directxmesh:x64-clangcl-dynamic
vcpkg install directxtex:x64-clangcl-dynamic
vcpkg install uvatlas:x64-clangcl-dynamic
vcpkg install directxtk:x64-clangcl-dynamic
vcpkg install directxtk12:x64-clangcl-dynamic

vcpkg install directxmesh:x64-clangcl-xboxone
vcpkg install directxtex:x64-clangcl-xboxone
vcpkg install uvatlas:x64-clangcl-xboxone
vcpkg install directxtk12:x64-clangcl-xboxone

vcpkg install directxmesh:x64-clangcl-scarlett
vcpkg install directxtex:x64-clangcl-scarlett
vcpkg install uvatlas:x64-clangcl-scarlett
vcpkg install directxtk12:x64-clangcl-scarlett
```

**Comment by walbourn** [comment]
Failure is some problem with another recent update.

**Comment by dg0yt** [comment]
> Failure is some problem with another recent update.

Are you sure?

~~~
-- Configuring done
CMake Error at D:/installed/x64-windows/share/directxtex/DirectXTex-targets.cmake:60 (set_target_properties):
  The link interface of target "Microsoft::DirectXTex" contains:

    OpenMP::OpenMP_CXX

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  D:/installed/x64-windows/share/directxtex/directxtex-config.cmake:27 (include)
  C:/a/1/s/scripts/buildsystems/vcpkg.cmake:852 (_find_package)
  src/CMakeLists.txt:109 (find_package)
~~~

**Comment by walbourn** [comment]
> > Failure is some problem with another recent update.
> 
> Are you sure?

You are quite right. Sorry, I didn't recognize ``rsa-bsa`` as a DirectXTex consumer. Looking.

**Comment by walbourn** [comment]
> > > Failure is some problem with another recent update.
> > 
> > 
> > Are you sure?
> 
> You are quite right. Sorry, I didn't recognize `rsa-bsa` as a DirectXTex consumer. Looking.

I see the issue. Now that I've switched to using ``find_package(OpenMP)`` to more robustly enable OpenMP, this is now a dependency for clients of the library. My question is for "Modern CMake" usage, should I:

(a) require that CMake clients of the library be updated to require:

```
find_package(OpenMP)
```

-or-

(b) Update directxtex-targets and uvatlas-targets files to include:

```
if(NOT TARGET OpenMP::OpenMP_CXX)
  find_package(OpenMP)
endif()
```

**Comment by walbourn** [comment]
Fixed CMake issues in upstream, so I removed the patches. This fixes the OpenMP targets issues as well.

**Pull request merged**